smartcontactmanager/src/main/java/com/scm/smartcontactmanager/service/category/CategoryServiceImpl.java

```diff
@@ -1,8 +1,137 @@
 package com.scm.smartcontactmanager.service.category;
 
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.HttpStatusCode;
 import org.springframework.stereotype.Service;
+import com.scm.smartcontactmanager.controller.category.request.CategoryRequestDto;
+import com.scm.smartcontactmanager.controller.category.request.ContactIds;
+import com.scm.smartcontactmanager.controller.category.response.CategoryResponseDto;
+import com.scm.smartcontactmanager.dao.category.CategoryDao;
+import com.scm.smartcontactmanager.entity.category.Category;
+import com.scm.smartcontactmanager.entity.contact.Contact;
+import com.scm.smartcontactmanager.entity.contactcategoryrelation.ContactCategoryRelation;
+import com.scm.smartcontactmanager.exceptionhandle.ServiceException;
+import com.scm.smartcontactmanager.repository.contact.ContactRespository;
+import com.scm.smartcontactmanager.repository.contactcategoryrelation.ContactCategoryRelationRepository;
+
+import jakarta.validation.Valid;
 
 @Service
-public class CategoryServiceImpl implements CategoryService{
+public class CategoryServiceImpl implements CategoryService {
+
+	@Autowired
+	private CategoryDao categoryDao;
+	@Autowired
+	private ContactCategoryRelationRepository contactCategoryRelationRepository ;
+
+	@Override
+	public CategoryResponseDto createCategory(@Valid CategoryRequestDto categoryRequestDto) {
+		Category category = mapToEntity(categoryRequestDto);
+		Category findCategory = categoryDao.findCategory(categoryRequestDto.getName(),categoryRequestDto.getType());
+		if(findCategory!=null) {
+			throw new ServiceException(HttpStatus.valueOf(409), "Category already exsists");
+		}else {
+			Category createCategory = categoryDao.createCategory(category);
```

Comment:
Variable name should be newCategory instead of createCategory

Reply:
Addressed